src/Console/Plugin/UninstallCommand.php

```diff
@@ -0,0 +1,235 @@
+    }
+
+    protected function getDirectoryChoiceQuestion()
+    {
+        return __('Which plugin(s) do you want to uninstall (comma separated values)?');
+    }
+
+    protected function getDirectoryChoiceChoices()
+    {
+        // Fetch directory list
+        $directories = [];
+        foreach (PLUGINS_DIRECTORIES as $plugins_directory) {
+            $directory_handle  = opendir($plugins_directory);
+            while (false !== ($filename = readdir($directory_handle))) {
+                if (
+                    !in_array($filename, ['.svn', '.', '..'])
+                    && is_dir($plugins_directory . DIRECTORY_SEPARATOR . $filename)
+                ) {
+                    $directories[] = $filename;
+                }
+            }
+        }
+
+        // Fetch plugins information
+        $choices = [];
+        foreach ($directories as $directory) {
+            $plugin = new Plugin();
+            $informations = $plugin->getInformationsFromDirectory($directory);
+
+            if (empty($informations)) {
+                continue; // Ignore directory if not able to load plugin information.
+            }
+
+            if ($this->isInstalled($directory)) {
+                continue;
+            }
+
+            $choices[$directory] = array_key_exists('name', $informations)
+                ? $informations['name']
+                : $directory;
+        }
+
+        ksort($choices, SORT_STRING);
+
+        return $choices;
+    }
+
+    /**
+     * Check if plugin is installed.
+     *
+     * @param string $directory
+     *
+     * @return array
+     */
+    private function isInstalled($directory)
+    {
+        $plugin = new Plugin();
+        $is_already_known = $plugin->getFromDBByCrit(['directory' => $directory]);
+        return $is_already_known && $plugin->fields['state'] != Plugin::NOTINSTALLED;
+    }
+
+    /**
+     * Check if uninstall method can be run for given plugin.
+     *
+     * @param string  $directory
+     *
+     * @return boolean
+     */
+    private function canRunUninstallMethod($directory)
+    {
+        $plugin = new Plugin();
+
+        // Check that directory is valid
+        $informations = $plugin->getInformationsFromDirectory($directory);
+        if (empty($informations)) {
+            $this->output->writeln(
+                '<error>' . sprintf(__('Invalid plugin directory "%s".'), $directory) . '</error>',
+                OutputInterface::VERBOSITY_QUIET
+            );
+            return false;
+        }
+
+        // Check if plugin is not already installed
+        if (!$this->isInstalled($directory)) {
+            $message = sprintf(
+                __('Plugin "%s" is not installed.'),
+                $directory
+            );
+            $this->output->writeln(
+                '<error>' . $message . '</error>',
+                OutputInterface::VERBOSITY_QUIET
+            );
+            return false;
+        }
```

Comment:
1. A check that plugin is already in DB should be done (see `$is_already_known = $plugin->getFromDBByCrit(['directory' => $directory]);` check in `ActivateCommand`). This is to handle an edge case: call to uninstall method of a plugin that is present on filesystem but not actually registered in DB.

2. Instead of checking `$this->isInstalled` result, a simple check on `$plugin->fields['state']` could be done to validate that status is one of `self::ACTIVATED, self::NOTUPDATED, self::TOBECONFIGURED, self::NOTACTIVATED` .